### PR TITLE
Improve design of ServerLoggerForwarder and ServerTestHost

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
contributes to https://github.com/microsoft/testfx/issues/2145

will followup with tests for the `ServerLoggerForwarderProvider`

### CODE REVIEW

File: src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    where T : class
+{
+    private readonly LogLevel _logLevel;
+
+    public InMemoryStoreLogger(LogLevel logLevel)
+    {
+        _logLevel = logLevel;
+    }
+
+    public ConcurrentBag<T> Values { get; private set; } = new();
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs]:
```suggestion
    public ConcurrentBag<T> Values { get; } = new();
```

Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs]: Shall it be only protected?

File: src/Platform/Microsoft.Testing.Platform/Logging/ServerLogMessageInMemoryStore.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.Logging;
+
+internal sealed class ServerLogMessageInMemoryStore : InMemoryStoreLogger<ServerLogMessage>
+{
+    public ServerLogMessageInMemoryStore(LogLevel logLevel)
+        : base(logLevel)
+    {
+    }
```
Comment (Evangelink) [inline src/Platform/Microsoft.Testing.Platform/Logging/ServerLogMessageInMemoryStore.cs]: Nit (not mandatory)

```suggestion
internal sealed class ServerLogMessageInMemoryStore(LogLevel logLevel) : InMemoryStoreLogger<ServerLogMessage>(logLevel)
{
```

**Review by Evangelink — DISMISSED** [review]

File: src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs]: mmm needs to be accessed from outside, what you mean?

Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs]: like `IEnum<T>` on the type and make it protected?

Comment (MarcoRossignoli) [reply src/Platform/Microsoft.Testing.Platform/Logging/InMemoryStoreLogger.cs]: We can also remove this type and implement it only for the server mode...maybe it's too much have this abstraction.

**Review by Evangelink — APPROVED** [review]

**Pull request merged**